tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx

```diff
@@ -271,20 +273,39 @@ BuildAction(const ColumnNames_t &colNames, const std::shared_ptr<SnapshotHelperA
       isDefine[i] = colRegister.IsDefineOrAlias(colNames[i]);
 
    std::unique_ptr<RActionBase> actionPtr;
-   if (!ROOT::IsImplicitMTEnabled()) {
-      // single-thread snapshot
-      using Helper_t = SnapshotHelper<ColTypes...>;
+   if (snapHelperArgs->fToNTuple) {
+#ifdef R__HAS_ROOT7
+      // TODO(fdegeus) Add MT snapshotting
+      using Helper_t = SnapshotRNTupleHelper<ColTypes...>;
       using Action_t = RAction<Helper_t, PrevNodeType>;
+
+      auto loopManager = snapHelperArgs->fLoopManager;
+
       actionPtr.reset(
-         new Action_t(Helper_t(filename, dirname, treename, colNames, outputColNames, options, std::move(isDefine)),
+         new Action_t(Helper_t(filename, treename, colNames, outputColNames, options, loopManager, std::move(isDefine)),
                       colNames, prevNode, colRegister));
+
+      return actionPtr;
+#else
+      throw std::runtime_error("Cannot snapshot to RNTuple: this installation of ROOT has not been build with ROOT7 "
+                               "components enabled.");
```

Comment:
We usually/try to prefix error messages in  RDF with `RDataFrame:`.

Reply:
Changed here and everywhere else :+1: